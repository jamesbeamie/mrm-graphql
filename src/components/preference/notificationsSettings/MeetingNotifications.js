import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import NotificationSetting from './NotificationSetting';
import UPDATE_NOTIFICATION_MUTATION from '../../../graphql/mutations/Preferences';

/**
 * Component that shows notifications related to meetings
 *
 * @returns {JSX}
 */
class MeetingNotifications extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    meeting: PropTypes.bool.isRequired,
  };
  state = {
    switchToggle: this.props.meeting,
  };

  render() {
    const { title, body } = this.props;
    const { switchToggle } = this.state;
    return (
      <Mutation mutation={UPDATE_NOTIFICATION_MUTATION}>
        {mutate => (
          <NotificationSetting
            title={title}
            body={body}
            toggle={switchToggle}
            handleChange={() => {
              const toggle = switchToggle;
              this.setState({ switchToggle: !toggle }, () => {
                mutate({ variables: { meeting: this.state.switchToggle } });
              });
            }}
          />
        )}
      </Mutation>
    );
  }
}

export default MeetingNotifications;
