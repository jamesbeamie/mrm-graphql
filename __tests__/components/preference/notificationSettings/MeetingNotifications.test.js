import React from 'react';
import { shallow, mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import UPDATE_NOTIFICATION_MUTATION from '../../../../src/graphql/mutations/Preferences';
import MeetingNotifications from '../../../../src/components/preference/notificationsSettings/MeetingNotifications';

describe('Toggle True Meeting Notifications Component', () => {
  const mocks = [
    {
      request: {
        query: UPDATE_NOTIFICATION_MUTATION,
        variables: { meeting: true },
      },
      result: {
        data: {
          updateNotification: {
            notification: {
              id: '1',
              userId: 17,
              meetingUpdateNotification: true,
              deviceHealthNotification: true,
            },
          },
        },
      },
    },
  ];

  const props = {
    title: 'Meeting Notifications',
    body: 'some',
    meeting: false,
  };
  const falseProps = {
    title: 'Meeting Notifications',
    body: 'some',
    meeting: true,
  };

  const setup = (
    <MockedProvider mocks={mocks} addTypename={false}>
      <MeetingNotifications {...props} />
    </MockedProvider>
  );

  const wrapper = mount(setup);

  it('Should toggle True', async () => {
    const switchBtn = wrapper.find('NotificationSetting');
    switchBtn.props().handleChange();
    await wait(0);
    wrapper.update();
    expect(wrapper.text()).toContain('Meeting Notificationssome');
  });

  it('Should toggle False', async () => {
    const falseMocks = [
      {
        request: {
          query: UPDATE_NOTIFICATION_MUTATION,
          variables: { meeting: false },
        },
        result: {
          data: {
            updateNotification: {
              notification: {
                id: '1',
                userId: 17,
                meetingUpdateNotification: true,
                deviceHealthNotification: false,
              },
            },
          },
        },
      },
    ];

    const falseSetup = (
      <MockedProvider mocks={falseMocks} addTypename={false}>
        <MeetingNotifications {...falseProps} />
      </MockedProvider>
    );
    const falseWrapper = mount(falseSetup);
    const switchBtn = falseWrapper.find('NotificationSetting');
    switchBtn.props().handleChange();
    await wait(0);
    falseWrapper.update();
    expect(falseWrapper.text()).toContain('Meeting Notificationssome');
  });

  it('renders correctly from memory', () => {
    expect(shallow(<MeetingNotifications {...props} />)).toMatchSnapshot();
  });
  it('renders correctly from memory again', () => {
    expect(shallow(<MeetingNotifications {...falseProps} />)).toMatchSnapshot();
  });
});
