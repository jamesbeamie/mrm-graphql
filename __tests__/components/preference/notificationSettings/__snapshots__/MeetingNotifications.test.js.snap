// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle True Meeting Notifications Component renders correctly from memory 1`] = `
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "updateNotification",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "meetingUpdateNotification",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "meeting",
                      },
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "deviceHealthNotification",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "device",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "updateNotification",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "notification",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "userId",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "meetingUpdateNotification",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "deviceHealthNotification",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Boolean",
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "meeting",
                },
              },
            },
            Object {
              "defaultValue": undefined,
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Boolean",
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "device",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 316,
        "start": 0,
      },
    }
  }
>
  <Component />
</Mutation>
`;

exports[`Toggle True Meeting Notifications Component renders correctly from memory again 1`] = `
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "updateNotification",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "meetingUpdateNotification",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "meeting",
                      },
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "deviceHealthNotification",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "device",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "updateNotification",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "notification",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "userId",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "meetingUpdateNotification",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "deviceHealthNotification",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Boolean",
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "meeting",
                },
              },
            },
            Object {
              "defaultValue": undefined,
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Boolean",
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "device",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 316,
        "start": 0,
      },
    }
  }
>
  <Component />
</Mutation>
`;
